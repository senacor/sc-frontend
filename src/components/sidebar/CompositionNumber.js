import React, { useEffect, useState } from 'react';
import Typography from '@material-ui/core/Typography';

import { withStyles } from '@material-ui/core/styles';
import { injectIntl } from 'react-intl';

const styles = theme => ({
  version: {
    padding: theme.spacing.unit
  }
});

const CompositionNumber = props => {
  const [composition, setComposition] = useState('unknown');

  useEffect(() => {
    const fetchData = async () => {
      try {
        const res = await fetch('/static/composition-number.json');
        setComposition(await res.json());
      } catch (err) {
        setComposition('localhost');
      }
    };
    fetchData();
  }, []);

  return (
    <div id="composition-number" className={props.classes.version}>
      <Typography color="textSecondary">
        {`${props.intl.formatMessage({
          id: 'compositionnumber.version'
        })} ${composition}`}
      </Typography>
    </div>
  );
};

export default injectIntl(withStyles(styles)(CompositionNumber));
