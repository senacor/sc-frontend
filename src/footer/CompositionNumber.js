import React, { Component } from 'react';
import Typography from 'material-ui/Typography';

class CompositionNumber extends Component {
  constructor(props) {
    super(props);
    this.state = {
      composition: 'unknown'
    };
  }

  async componentDidMount() {
    try {
      const res = await fetch('/static/composition-number.json');

      this.setState({
        composition: await res.json()
      });
    } catch (err) {
      this.setState({
        composition: 'failed'
      });
    }
  }

  render() {
    return (
      <div id="composition-number">
        <Typography>current composition: {this.state.composition}</Typography>
      </div>
    );
  }
}

export default CompositionNumber;
