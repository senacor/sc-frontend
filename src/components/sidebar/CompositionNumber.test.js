import React from 'react';
import CompositionNumber from './CompositionNumber';
import { shallow } from 'enzyme';

it('displays the composition number', async () => {
  global.fetch = jest.fn();
  const fetchRes = Promise.resolve({
    json: () => Promise.resolve(1)
  });
  fetch.mockReturnValueOnce(fetchRes);

  const component = shallow(<CompositionNumber />);

  // we need to wait, that the response is processed in the component
  const res = await fetchRes;
  await res.json();
  component.update();
  expect(component).toMatchSnapshot();
});

it('displays failure if the resource is not found', async () => {
  global.fetch = jest.fn();
  const fetchRes = Promise.resolve({
    status: 404
  });
  fetch.mockReturnValueOnce(fetchRes);

  const component = shallow(<CompositionNumber />);

  // we need to wait, that the response is processed in the component
  await fetchRes;
  component.update();
  expect(component).toMatchSnapshot();
});
