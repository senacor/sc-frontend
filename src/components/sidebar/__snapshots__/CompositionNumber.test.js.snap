// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays failure if the resource is not found 1`] = `
<CompositionNumber
  classes={
    Object {
      "version": "CompositionNumber-version-1",
    }
  }
/>
`;

exports[`displays the composition number 1`] = `
<CompositionNumber
  classes={
    Object {
      "version": "CompositionNumber-version-1",
    }
  }
/>
`;
